glib/src/value.rs

```diff
@@ -930,6 +936,33 @@ impl SetValueOptional for str {
     }
 }
 
+impl<'a> FromValue<'a> for Value {
+    unsafe fn from_value(value: &'a Value) -> Self {
+        from_glib_none(
+            gobject_ffi::g_value_get_boxed(value.to_glib_none().0) as *mut gobject_ffi::GValue
+        )
+    }
+}
+
+impl<'a> FromValueOptional<'a> for Value {
+    unsafe fn from_value_optional(value: &'a Value) -> Option<Self> {
+        Some(FromValue::from_value(value))
+    }
+}
+
+impl SetValue for Value {
+    unsafe fn set_value(value: &mut Value, this: &Self) {
+        gobject_ffi::g_value_set_boxed(value.to_glib_none_mut().0, this.to_glib_full() as _)
```

Comment:
```suggestion
        gobject_ffi::g_value_set_boxed(value.to_glib_none_mut().0, this.to_glib_none().0 as ffi::gpointer)
```

Otherwise you leak memory: `full` creates a copy, `set_boxed` does not take ownership and also creates a copy